redwood-lazylayout-view/src/main/kotlin/app/cash/redwood/lazylayout/view/ViewLazyList.kt

```diff
@@ -101,9 +104,11 @@ internal open class ViewLazyList private constructor(
       addOnScrollListener(
         object : RecyclerView.OnScrollListener() {
           override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
+            val viewsOnScreen = recyclerView.children.map { recyclerView.getChildViewHolder(it).bindingAdapterPosition }
```

Comment:
This is a decently expensive operation to perform on every frame of scrolling. It involves allocation of the list and boxing of the position integers.

Reply:
It is.

There was an alternative of passing in a mutable of viewHolder and positions to adapter to keep track of positions of visible views (by adding/removing keys when `onViewDetachedFromWindow` and `onBindViewHolder`), it's somewhat ugly though.

There might be other alternatives. 

Previous calls : `linearLayoutManager.findFirstVisibleItemPosition()` and `linearLayoutManager.findLastVisibleItemPosition()` are not giving the accurate indices.

Reply:
> Previous calls : `linearLayoutManager.findFirstVisibleItemPosition()` and `linearLayoutManager.findLastVisibleItemPosition()` are not giving the accurate indices.

I'd be curious to investigate this more, because it's kinda surprising that a core API would fail like this. Did you find any bug reports on it?

Reply:
> Did you find any bug reports on it?

from my time-boxed googling, there wasn't report with similar symptom